nextjs-v12/e2e/checkNextJSv12Apps.cy.ts

```diff
@@ -0,0 +1,117 @@
+import {BaseMethods} from "../../cypress/common/base";
+import {baseSelectors} from "../../cypress/common/selectors";
+import {Constants} from "../../cypress/fixtures/constants";
+const basePage: BaseMethods = new BaseMethods()
+
+const appsData = [
+    {
+        appName: 'Checkout',
+        host: 3000
+    },
+    {
+        appName: 'Home',
+        host: 3001
+    },
+    {
+        appName: 'Shop',
+        host: 3002
+    }
+]
+
+appsData.forEach((
+    property: {
+        appName: string,
+        host: number
+    }
+) => {
+    describe(`Check ${property.appName} app elements and links are visible and not broken`, () => {
+        beforeEach(() => {
+            basePage.openLocalhost({
+                number: property.host,
+                wait: 2000
+            })
+        })
+
+        it (`Check application functionality and elements`, () => {
```

Comment:
have no idea what is going on here. Better to split to multiple its with proper names